tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,144 @@
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MongoDB.Driver.Tests
+{
+    public class OcspIntegrationTests
+    {
+        private static readonly string _ocspOutcomeEnvironmentVariableName = "OCSP_TLS_SHOULD_SUCCEED";
+        private readonly ITestOutputHelper _output;
+
+        public OcspIntegrationTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        /*
+         * This test must be run as its own task in EG.
+         * When testing locally, this test should not run alongside other integration tests.
+         * Clearing local OCSP caches between test runs is also required when testing locally.
+         * Clearing the OCSP caches may involve clearing out the caches in ~\.dotnet folder
+         * and $DOTNET_CLI_HOME/.dotnet, as well as running `dotnet clean` prior to running the tests.
+         * See spec for OCSP OS-level cache manipulation commands.
+         *   https://github.com/mongodb/specifications/blob/master/source/ocsp-support/ocsp-support.rst#os-level-ocsp-cache-manipulation
+         * When testing on Windows, the certificate should be added to the trust store prior to each run in order to
+         * reduce the chances of Windows pruning the certificate from the trust store prior to the test running.
+         */
+        [SkippableFact]
+        public void MongoClientShouldRespectCertificateStatusAndTlsInsecure()
+        {
+            /* We cannot call RequireServer.Check() because this would result in a connection being made to the mongod
+             * and that connection will not succeed if we're testing the revoked certificate case. */
+
+            // Uncommenting the line below may help when testing locally
+            // SetupLocalTestingEnvironmentVariables(mockResponderValidatesAllCertificates: false);
+            var secureClientShouldConnect = GetSecureClientExpectedOutcomeOrSkipTest();
+            /* To prevent OCSP caching from polluting the test results, we MUST run the "secure" version before
+             * the tlsInsecure version.  */
+            var secureClientException = Record.Exception(() => Ping());
+            var tlsInsecureClientException = Record.Exception(()=> Ping(tlsInsecure: true));
+
+            tlsInsecureClientException.Should().BeNull();
+            if (secureClientShouldConnect)
+            {
+                secureClientException.Should().BeNull();
+            }
+            else
+            {
+                secureClientException.Should().BeOfType<TimeoutException>();
+                var message = secureClientException.Message;
+                // The exception will lack this message if the heartbeat doesn't fire
+                message.Should().Contain("The remote certificate is invalid according to the validation procedure");
+            }
+
+            void Ping(bool tlsInsecure = false)
```

Comment:
I don't think I'd use an optional parameter in a local method that is only called twice.

Might as well be explicit.